src/model.cpp

```diff
@@ -1928,10 +1967,38 @@ void Model::fdJrzdsigma(const int ie, const int nroots, const realtype t,
     }
 }
 
+void Model::fspl(const realtype t) {
+    for (int ispl = 0; ispl < nspl; ispl++) {
+        if (splines_[ispl].get_logarithmic_parametrization()) {
+            state_.spl_[ispl] = std::exp(splines_[ispl].getValue(t));
+        } else {
+            state_.spl_[ispl] = splines_[ispl].getValue(t);
+        }
```

Comment:
Can/should that `if` not go into `getValue`?

Reply:
@lcontento ?

Reply:
I think it can and I think it should, not sure if there is any reason why Paul put it outside.
I can try putting inside `getValue` and see if anything breaks.